File: languages/csharp/01-foundations/arrays-and-vectors/example/main.cs
// Module focus: Storing related values in ordered collections and iterating safely.
// Why it matters: the example makes it possible to store and traverse ordered collections safely
// before the learner tackles the exercises.

using System;
using System.Collections.Generic;

// Separate helpers keep the main path focused on how to store and traverse ordered collections
// safely.
class Program
{
    // Fixed inputs make the consequence of trusting collection size input when count is zero or
    // negative visible and repeatable.
    static void Main()
    {
        // These values exercise the normal path before the exercises vary the documented
        // boundaries.
        int[] fixedScores = { 72, 88, 95 };
        // The printed result shows whether the program can handle empty collections and index
        // boundaries explicitly.
        Console.WriteLine($"Fixed array values: {string.Join(", ", fixedScores)}");

        Console.Write("How many temperatures do you want to enter? ");
        int count = int.Parse(Console.ReadLine() ?? "0");

        if (count <= 0)
        {
            Console.WriteLine("Nothing to process.");
            return;
        }

        List<double> temperatures = new List<double>();
        for (int i = 0; i < count; i++)
        {
            Console.Write($"Temperature {i + 1}: ");
            temperatures.Add(double.Parse(Console.ReadLine() ?? "0"));
        }

        double sum = 0;
        foreach (double value in temperatures)
        {
            sum += value;
        }

        Console.WriteLine($"Average temperature: {sum / temperatures.Count:F2}");
    }
}

File: languages/go/01-foundations/arrays-and-vectors/example/main.go
// Module focus: Storing related values in ordered collections and iterating safely.
// Why it matters: the example makes it possible to store and traverse ordered collections safely
// before the learner tackles the exercises.

package main

import "fmt"

// Fixed inputs make the consequence of trusting collection size input when count is zero or
// negative visible and repeatable.
func main() {
	// These values exercise the normal path before the exercises vary the documented boundaries.
	fixedScores := [3]int{72, 88, 95}
	// The printed result shows whether the program can handle empty collections and index boundaries
	// explicitly.
	fmt.Printf("Fixed array values: %d, %d, %d\n", fixedScores[0], fixedScores[1], fixedScores[2])

	var count int
	fmt.Print("How many temperatures do you want to enter? ")
	fmt.Scanln(&count)

	if count <= 0 {
		fmt.Println("Nothing to process.")
		return
	}

	temperatures := make([]float64, 0, count)
	for i := 0; i < count; i++ {
		var value float64
		fmt.Printf("Temperature %d: ", i+1)
		fmt.Scanln(&value)
		temperatures = append(temperatures, value)
	}

	sum := 0.0
	for _, value := range temperatures {
		sum += value
	}

	average := sum / float64(len(temperatures))
	fmt.Printf("Average temperature: %.2f\n", average)
}
